tests/test_modeling_common.py

```diff
@@ -4829,6 +4831,73 @@ def test_forward_with_num_logits_to_keep(self):
             # Assert the last tokens are actually the same (except for the natural fluctuation due to order of FP ops)
             self.assertTrue(torch.allclose(all_logits[:, -1:, :], last_token_logits, atol=1e-5))
 
+    @slow
+    @require_torch_greater_or_equal("2.3")
+    def test_torch_export(self, config_kwargs=None):
+        if not self.test_torch_exportable:
+            self.skipTest(reason="Torch export test is not enabled for this model.")
+
+        def recursively_check(eager_outputs, exported_outputs):
+            is_tested = False
+            if isinstance(eager_outputs, torch.Tensor):
+                self.assertEqual(eager_outputs.shape, exported_outputs.shape)
+                self.assertTrue(torch.allclose(eager_outputs, exported_outputs, atol=1e-4))
+                return True
+            elif isinstance(eager_outputs, (tuple, list)):
+                for eager_output, exported_output in zip(eager_outputs, exported_outputs):
+                    is_tested = is_tested or recursively_check(eager_output, exported_output)
+                return is_tested
+            return is_tested
```

Comment:
The idea with True/False is to check if we test anything at all. For example, if all `outputs` are not of list/Tensor type or if `outputs` is an empty list. That is why we have `or` here, we need at least one check to be true. In case there is a difference between eager and compiled outputs, an error will be raised, and it is not related to the is_tested = True/False flag.

Reply:
Got it! Maybe having this as a comment in the code is nice.